flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,14 +191,41 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
+
+		while(true) {
+			GetClusterStatusResponse status = getClusterStatus();
+			if (status != null) {
+				if (status.numRegisteredTaskManagers() < clusterDescriptor.getTaskManagerCount()) {
+					logAndSysout("TaskManager status (" + status.numRegisteredTaskManagers() + "/"
+						+ clusterDescriptor.getTaskManagerCount() + ")");
+				} else {
+					logAndSysout("All TaskManagers are connected");
+					break;
+				}
+			} else {
+				logAndSysout("No status updates from the YARN cluster received so far. Waiting ...");
+			}
+
+			try {
+				Thread.sleep(500);
+			}
+			catch (InterruptedException e) {
+				LOG.error("Interrupted while waiting for TaskManagers");
+				System.err.println("Thread is interrupted");
+				Thread.currentThread().interrupt();
```

Comment:
This is the old code that was used for deployment. Though at a different place. Would be good to make it interruptible and cleanup everything.